scheduler/task_finder_test.go

```diff
@@ -16,84 +16,277 @@ func init() {
+	// XXX: collection is deliberately empty
+	runnableTasks, err := s.taskFinder.FindRunnableTasks()
+	s.Nil(err)
+	s.Empty(runnableTasks)
+}
+
+func (s *TaskFinderSuite) TestInactiveTasksNeverReturned() {
+	// insert the tasks, setting one to inactive
+	s.tasks[2].Activated = false
+	s.insertTasks()
+
+	// finding the runnable tasks should return two tasks
+	runnableTasks, err := s.taskFinder.FindRunnableTasks()
+	s.Nil(err)
+	s.Len(runnableTasks, 2)
+}
+
+func (s *TaskFinderSuite) TestTasksWithUnsatisfiedDependenciesNeverReturned() {
+	// edit the dependency tasks, setting one to have finished
+	// successfully and one to have finished unsuccessfully
+	s.depTasks[0].Status = evergreen.TaskFailed
+	s.depTasks[1].Status = evergreen.TaskSucceeded
+
+	// edit the tasks, setting one to have unmet dependencies, one to
+	// have no dependencies, and one to have successfully met
+	// dependencies
+	s.tasks[0].DependsOn = []task.Dependency{}
+	s.tasks[1].DependsOn = []task.Dependency{{s.depTasks[0].Id, evergreen.TaskSucceeded}}
+	s.tasks[2].DependsOn = []task.Dependency{{s.depTasks[1].Id, evergreen.TaskSucceeded}}
+
+	s.insertTasks()
+
+	// finding the runnable tasks should return two tasks (the one with
+	// no dependencies and the one with successfully met dependencies
+	runnableTasks, err := s.taskFinder.FindRunnableTasks()
+	s.Nil(err)
+	s.Len(runnableTasks, 2)
+}
+
+type TaskFinderComparisonSuite struct {
+	suite.Suite
+	oldRunnableTasks []task.Task
+	newRunnableTasks []task.Task
+}
+
+func TestTaskFinderComparisonSuite(t *testing.T) {
+	s := new(TaskFinderComparisonSuite)
+
+	suite.Run(t, s)
+}
+
+func (s *TaskFinderComparisonSuite) SetupTest() {
+	session, _, _ := db.GetGlobalSessionFactory().GetSession()
+	s.NotNil(session)
+	s.NoError(session.DB(taskFinderTestConf.Database.DB).DropDatabase())
+
+	for _, task := range tasks {
+		s.NoError(task.Insert())
+	}
+
+	oldTaskFinder := &LegacyDBTaskFinder{}
+	newTaskFinder := &DBTaskFinder{}
+	var (
+		err_old error = nil
+		err_new error = nil
+	)
+
+	s.oldRunnableTasks, err_old = oldTaskFinder.FindRunnableTasks()
+	s.Nil(err_old)
+	s.newRunnableTasks, err_new = newTaskFinder.FindRunnableTasks()
+	s.Nil(err_new)
+}
+
+func (s *TaskFinderComparisonSuite) TestFindRunnableHostsIsIdentical() {
+	s.NotEmpty(s.oldRunnableTasks)
+	s.NotEmpty(s.newRunnableTasks)
+
+	s.Equal(s.oldRunnableTasks, s.newRunnableTasks)
+}
+
+func (s *TaskFinderComparisonSuite) TestExpectedRunnableHosts() {
+	idsOldMethod := []string{}
+	for _, task := range s.oldRunnableTasks {
+		idsOldMethod = append(idsOldMethod, task.Id)
+	}
+
+	idsNewMethod := []string{}
+	for _, task := range s.newRunnableTasks {
+		idsNewMethod = append(idsNewMethod, task.Id)
+	}
+
+	expectedIds := []string{
+		"parent0-child1", "parent0-child2", "parent1-child0", "parent4",
+	}
+	s.Equal(idsOldMethod, expectedIds)
+	s.Equal(idsNewMethod, expectedIds)
+
+	s.Len(idsOldMethod, 4)
+	s.Len(idsNewMethod, 4)
```

Comment:
I am totally guilty of doing tests that main mechanism of enforcement is asserting the length of things are equal, as it's sort of opaque and is really a proxy for something else (in this case, eqaulity) but also in this case you're asserting that they're equal just above it so it's fine!

Equality implies the same order of tasks, so I guess my remaining concern is that this task assumes some ordering constraints that may not be true. Can we have a less order dependent test implementation?